pkgs/by-name/pi/pinchflat/package.nix

```diff
@@ -0,0 +1,70 @@
+{
+  lib,
+  fetchFromGitHub,
+  fetchYarnDeps,
+  beamPackages,
+  yarn,
+  nodejs,
+  esbuild,
+  tailwindcss,
+  fixup-yarn-lock,
+  apprise,
+  yt-dlp,
+  ...
+}:
+beamPackages.mixRelease rec {
+  pname = "pinchflat";
+  version = "2025.3.6";
+  src = fetchFromGitHub {
+    owner = "kieraneglin";
+    repo = "pinchflat";
+    rev = "v${version}";
+    hash = "sha256-hBYP8oCbaMjRTaXRK3tnTWN1nI+ADq3RFlb6c2t4kxo=";
+  };
+
+  mixNixDeps = import ./mix.nix {
+    inherit beamPackages lib;
+  };
+  removeCookie = false;
+
+  runtimeDeps = [
+    apprise
+    yt-dlp
+  ];
+
+  yarnOfflineCache = fetchYarnDeps {
+    yarnLock = "${src}/assets/yarn.lock";
+    sha256 = "sha256-xJL+qcohtu+OmZ31E1QU9uqBWAFGejKIO3XRd+R6z/4=";
+  };
+
+  nativeBuildInputs = [
+    fixup-yarn-lock
+    tailwindcss
+    yarn
+  ];
+  buildInputs = [ nodejs ];
+
+  postBuild = ''
+    export HOME=$PWD
+    fixup-yarn-lock ~/assets/yarn.lock
+    yarn --cwd assets config --offline set yarn-offline-mirror $yarnOfflineCache
+    yarn --cwd assets install --offline --frozen-lockfile --ignore-engines --ignore-scripts --no-progress
+    patchShebangs ~/assets/node_modules
+
+    # phoenixframework expects platform-specific tailwind/esbuild binaries in a specific location:
+    # https://github.com/phoenixframework/tailwind/blob/194ab0f979782e4ccf2fe796042bf8e20967df93/lib/tailwind.ex#L243-L251
+    ln -s "${tailwindcss}/bin/tailwindcss" "_build/tailwind-linux-x64"
+    ln -s "${esbuild}/bin/esbuild" "_build/esbuild-linux-x64"
+
+    mix do deps.loadpaths --no-deps-check, tailwind default --minify + esbuild default --minify + phx.digest
+  '';
+
+  meta = {
+    description = "Your next YouTube media manager";
+    homepage = "https://github.com/kieraneglin/pinchflat";
+    license = lib.licenses.agpl3Only;
+    maintainers = with lib.maintainers; [ charludo ];
+    platforms = [ "x86_64-linux" ];
```

Comment:
Does upstream only support x86_64?

Reply:
Not sure. I added only `x86_64-linux` since that's the only platform I am actually able to test on. Do you want me to add arm/mac targets regardless?

Reply:
I would specify `platforms.unix`. ofborg will build for x64/ARM Linux/Darwin, and I can test for both Linux architectures. And some basic tests on Darwin.

Reply:
Oh, awesome, thank you! 🎉